### DISCUSSION TITLE
no possibiltles to replace an asset

### DISCUSSION THREAD

**Discussion by angrotheman** [craftcms/cms#8643] [Ideas]
### Description

If i upload a file like 1.jpg and remove the asset from this field and upload a new 1.jpg i don't get asked if i want to overwriting the file or keep both but if i open the library and do the same thing then there is a pop up where i can decide what i would like to do.

### Steps to reproduce

1. Create a asset field
2. Add via drag-and-drop a file 
3. remove this asset from this field
4. Add via drag-and-drop a new file with the same file name.

The result will be something like file_12345.jpg

**Comment by brandonkelly** [comment]
This is expected behavior.

**Comment by angrotheman** [comment]
No. The expected behavior would be the same as uploading it to the file manager. With a pop-up which ask me if i want to replace the file.
My client want to change some pdfs and he has to remove all old pdfs from the file manager just to uploading the new pdf (with the same file name of course) via drag and drop.

**Comment by samhernandez** [comment]
@angrotheman Craft does allow for direct file replacement: https://monosnap.com/file/F4lDaqewPveK84jYhKsBK6TcfqDoim

I've trained around it with clients successfully helping them understand that file uploads are different beast than moving files around on a local filesystem.

**Comment by brandonkelly** [comment]
I can mark this as an enhancement request but it is definitely working as expected right now.

**Comment by angrotheman** [comment]
Ok then an enhancement. But for me and for my client too is this not logical.

**Comment by angrotheman** [comment]
> @angrotheman Craft does allow for direct file replacement: https://monosnap.com/file/F4lDaqewPveK84jYhKsBK6TcfqDoim
> 
> I've trained around it with clients successfully helping them understand that file uploads are different beast than moving files around on a local filesystem.

But this is another thing. If you upload a file via drag and drop and remove it later and upload then a new file with the same name, you have no option to replace the file.

**Comment by samhernandez** [comment]
@angrotheman If you're saying that the "Replace File" option is missing from a file, as in the screenshot I posted, then that sounds like a bug. That is as long as the user's permissions on that Volume allow them to upload and remove files. So if that's what's happening would you mind posting a bug report with steps to produce?

**Comment by angrotheman** [comment]
Ok maybe i am a bit unclear, so here is what I mean: 
![Bildschirmfoto 2019-03-12 um 23 04 00](https://user-images.githubusercontent.com/44032204/54239366-45dc2680-451b-11e9-97ad-70eb3902a164.png)
 I can drag and drop my file to "Add an asset".
![Bildschirmfoto 2019-03-12 um 23 06 52](https://user-images.githubusercontent.com/44032204/54239456-889dfe80-451b-11e9-8308-bdc56af1ff8c.png)
Then it is looking like this. 

If i now click the "x" to remove the file it is looking again like on the first image but if i now drag and drop a new file with the same name like before its automatically rename the file and there is no option to replace it.

I have to click on "Add an asset".
![Bildschirmfoto 2019-03-12 um 23 04 19](https://user-images.githubusercontent.com/44032204/54239562-cbf86d00-451b-11e9-9670-594077cc85c0.png)

Here i can add a new file with the same name and this pops out:
![Bildschirmfoto 2019-03-12 um 23 09 35](https://user-images.githubusercontent.com/44032204/54239696-390c0280-451c-11e9-8d46-798e23dad003.png)

It is taking a lot of more time if you want to change 50 pdfs for example
